code/modules/surgery/organs/internal/cyberimp/augments_arms.dm

```diff
@@ -1,17 +1,49 @@
-/obj/item/organ/cyberimp/arm
+/obj/item/organ/cyberimp/arm/
 	name = "arm-mounted implant"
-	desc = "You shouldn't see this! Adminhelp and report this as an issue on github!"
+	desc = "An implant that goes in your arm to improve it."
 	zone = BODY_ZONE_R_ARM
 	slot = ORGAN_SLOT_RIGHT_ARM_AUG
-	icon_state = "toolkit_generic"
 	w_class = WEIGHT_CLASS_SMALL
-	actions_types = list(/datum/action/item_action/organ_action/toggle)
 	valid_zones = list(
 		BODY_ZONE_R_ARM = ORGAN_SLOT_RIGHT_ARM_AUG,
 		BODY_ZONE_L_ARM = ORGAN_SLOT_LEFT_ARM_AUG,
 	)
 	///A ref for the arm we're taking up. Mostly for the unregister signal upon removal
 	var/obj/hand
+
+/obj/item/organ/cyberimp/arm/get_overlay_state(image_layer, obj/item/bodypart/limb)
+	return "[aug_overlay][zone == BODY_ZONE_L_ARM ? "_left" : "_right"]"
+
+/obj/item/organ/cyberimp/arm/on_mob_insert(mob/living/carbon/arm_owner)
+	. = ..()
+	RegisterSignal(arm_owner, COMSIG_CARBON_POST_ATTACH_LIMB, PROC_REF(on_limb_attached))
+	on_limb_attached(arm_owner, arm_owner.hand_bodyparts[zone == BODY_ZONE_R_ARM ? RIGHT_HANDS : LEFT_HANDS])
+
+/obj/item/organ/cyberimp/arm/on_mob_remove(mob/living/carbon/arm_owner)
+	. = ..()
+	on_limb_detached(hand)
+
+/obj/item/organ/cyberimp/arm/proc/on_limb_attached(mob/living/carbon/source, obj/item/bodypart/limb)
+	SIGNAL_HANDLER
+	if(!limb || QDELETED(limb) || limb.body_zone != zone)
+		return
+	if(hand)
+		on_limb_detached(hand)
+	RegisterSignal(limb, COMSIG_BODYPART_REMOVED, PROC_REF(on_limb_detached))
+	hand = limb
+
+/obj/item/organ/cyberimp/arm/proc/on_limb_detached(obj/item/bodypart/source)
+	SIGNAL_HANDLER
+	if(source != hand || QDELETED(hand))
+		return
+	UnregisterSignal(hand, list(COMSIG_BODYPART_REMOVED, COMSIG_ITEM_ATTACK_SELF))
```

Comment:
COMSIG_ITEM_ATTACK_SELF unreg should be on the toolkit subtype regardless of the fact that this technically works